.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,120 @@
+---
+name: CI/CD
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+  workflow_dispatch:
+
+env:
+  FORCE_COLOR: "1" # Make tools pretty.
+  PIP_DISABLE_PIP_VERSION_CHECK: "1"
+  PIP_NO_PYTHON_VERSION_WARNING: "1"
+  PYTHON_LATEST: "3.11"
+
+  # For re-actors/checkout-python-sdist
+  sdist-artifact: python-package-distributions
+
+permissions:
+  contents: read
+
+jobs:
+  build-sdist:
+    name: 📦 Build the source distribution
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+        name: Checkout project
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-python@v4
+        name: Setup ${{ env.PYTHON_LATEST }}
+        with:
+          python-version: ${{ env.PYTHON_LATEST }}
+      - run: python -Im pip install build
```

Comment:
Is there a particular reason to use `-I` (isolate from user envs) option here? I think it might be better to just allow GHAction's potential customization of Python environments.

Reply:
No idea, I've never personally used `-I` in my build chains, the original maintainer was adamant about having `-I` being included.

Reply:
I wish I documented why I included this, but I think we should remove `-I` if the reasoning cited is correct.

Reply:
Nowadays I tend to default to `-I` regardless of whether a command would be influenced...